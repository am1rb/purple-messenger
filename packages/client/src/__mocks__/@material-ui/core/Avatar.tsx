import React from "react";
import type { AvatarProps } from "@material-ui/core/Avatar";

function MuiAvatar({ src, alt }: AvatarProps) {
  return <img data-testid="mock-mui-avatar" src={src} alt={alt} />;
}

export default MuiAvatar;
