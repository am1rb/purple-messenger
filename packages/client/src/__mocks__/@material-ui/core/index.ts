export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Card } from "./Card";
export { default as CardContent } from "./CardContent";
export { default as Typography } from "./Typography";
export { default as Button } from "./Button";
