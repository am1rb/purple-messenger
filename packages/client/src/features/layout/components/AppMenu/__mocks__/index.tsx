import React from "react";

const appMenu = jest.fn(() => <div data-testid="app-menu" />);

export default appMenu;
