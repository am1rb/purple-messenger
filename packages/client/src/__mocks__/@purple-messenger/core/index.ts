import * as purpleMessengerCore from "@purple-messenger/core";

const mockedPurpleMessengerCore = jest.genMockFromModule<
  typeof purpleMessengerCore
>("@purple-messenger/core");

function mockedAction(action = () => ({ type: "" })) {
  return jest.fn(action);
}

module.exports = {
  ...mockedPurpleMessengerCore,
  loadConversationList: mockedAction(),
  unloadConversationList: mockedAction(),
  signOut: mockedAction(),
  clearMessageList: mockedAction(),
  seenMessageAck: mockedAction(),
  start: mockedAction(),
  shutdown: mockedAction(),
  submitMessage: mockedAction(),
  startTypingMessage: mockedAction(),
  stopTypingMessage: mockedAction(),
  signIn: mockedAction(),
  setAuthError: mockedAction(),
  verifyToken: mockedAction(),
  connected: mockedAction(purpleMessengerCore.connected),
  disconnected: mockedAction(purpleMessengerCore.disconnected),
};
