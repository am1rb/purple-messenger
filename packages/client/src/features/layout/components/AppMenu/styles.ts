import { makeStyles, Theme } from '@material-ui/core';

export default makeStyles((theme: Theme) => ({
  root: {
    width: 124,
    backgroundColor: '#292b4b',
    paddingTop: 15,
  },
  firstName: { color: theme.palette.background.paper },
}));
