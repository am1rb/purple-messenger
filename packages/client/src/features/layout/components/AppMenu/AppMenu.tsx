import React, { memo, useCallback } from "react";
import clsx from "clsx";
import { useDispatch, useSelector } from "react-redux";
import { Button, Typography, Box } from "@material-ui/core";
import { signOut } from "@purple-messenger/core";
import UserAvatar from "features/profile/components/UserAvatar";
import { getProfileInfo } from "features/profile/selectors";
import useStyles from "./AppMenu.styles";

export interface AppMenuProps {
  className?: string;
}

function AppMenu({ className }: AppMenuProps) {
  const classes = useStyles();
  const dispatch = useDispatch();
  const profile = useSelector(getProfileInfo);

  const handleSignOut = useCallback(() => {
    dispatch(signOut());
  }, [dispatch]);

  return (
    <div className={clsx(classes.root, className)}>
      <UserAvatar />
      <Box textAlign="center" mt={2}>
        <Typography className={classes.firstName} color="inherit" gutterBottom>
          Hi, {profile?.firstName}
        </Typography>
        <Button
          color="secondary"
          onClick={handleSignOut}
          variant="outlined"
          size="small"
        >
          signOut
        </Button>
      </Box>
    </div>
  );
}

export default memo(AppMenu);
