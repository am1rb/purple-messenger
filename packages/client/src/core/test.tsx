import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ConnectedRouter } from "connected-react-router";
import configureStore, { history } from "core/redux/configureStore";
import { State as RootState } from "core/redux/reducers";
import theme from "core/style/theme";
import { ThemeProvider } from "@material-ui/styles";

export function renderWithStore(
  ui: React.ReactElement,
  initStore: Partial<RootState>
) {
  const store = configureStore(initStore);

  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>{children}</ConnectedRouter>
        </Provider>
      </ThemeProvider>
    ),
  });
}
