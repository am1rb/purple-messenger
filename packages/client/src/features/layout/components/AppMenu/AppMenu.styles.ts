import { makeStyles, Theme } from "@material-ui/core/styles";

export default makeStyles((theme: Theme) => ({
  root: {
    width: 124,
    backgroundColor: "#292b4b",
    paddingTop: theme.spacing(2),
  },
  firstName: { color: theme.palette.background.paper },
}));
