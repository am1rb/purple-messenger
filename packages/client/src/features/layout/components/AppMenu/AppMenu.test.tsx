import React from "react";
import fireEvent from "@testing-library/user-event";
import { sampleProfileInfo, signOut } from "@purple-messenger/core";
import { renderWithStore, InitialStore } from "core/test";
import AppMenu from "./AppMenu";

const sharedStore: InitialStore = {
  profile: sampleProfileInfo,
};

describe("The <AppMenu /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = renderWithStore(<AppMenu />, sharedStore);
    expect(container).toMatchSnapshot();
  });

  it("Should call signOut action when the button is clicked", () => {
    const { getByText, store } = renderWithStore(<AppMenu />, sharedStore);
    fireEvent.click(getByText("signOut"));
    expect(store.actions).toContainEqual(signOut());
  });
});
