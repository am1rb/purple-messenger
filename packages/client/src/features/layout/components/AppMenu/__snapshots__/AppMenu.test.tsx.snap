// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <AppMenu /> tests Should match the snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <img
      data-testid="mock-mui-avatar"
      src="https://picsum.photos/200"
    />
    <div
      data-testid="mock-mui-box"
    >
      <div
        data-testid="mock-mui-typography"
      >
        Hi, 
        Logan
      </div>
      <button
        data-testid="mock-mui-button"
      >
        signOut
      </button>
    </div>
  </div>
</div>
`;
